src/ray/object_manager/pull_manager.h

```diff
@@ -68,7 +71,15 @@ class PullManager {
   ///
   /// \return True if a pull was cancelled. If there was no pending pull request for the
   /// object this method may return false.
-  bool CancelPull(const ObjectID &object_id);
+  void CancelPull(uint64_t request_id, std::vector<ObjectID> *objects_to_cancel);
+
+  /// Try to fetch an object immediately, by restoring the object from external
+  /// storage or by fetching the object from one of its expected client
+  /// locations.
+  ///
+  /// \param object_id The object's object id.
+  /// \return Void.
+  void TryFetch(const ObjectID &object_id);
```

Comment:
Why do we need this extra method? Can it be removed?

Reply:
I added it to trigger a pull for an object that was pulled successfully, then got evicted. It's not strictly necessary, but it can add delays when there is thrashing because we'd have to wait for the next PullManager tick before fetching again.

Reply:
Can we drop this for now? I don't think waiting a tick is a big deal considering thrashing is already the slow slow path.